src/daemon/distribution.c

```diff
@@ -340,48 +341,91 @@ double distribution_squared_deviation_sum(distribution_t *dist) {
   return squared_deviation_sum;
 }
 
-/* TODO(bkjg): add tests for this function */
-int distribution_sub(distribution_t *d1, distribution_t *d2) {
+static int compare_longint(uint64_t a, uint64_t b) {
+  return a == b ? 0 : a < b ? -1 : 1;
+}
+
+/** return a pointer to array int[2].
+ * The first value is error code or 0 if succeed.
+ * The second value is result of comparison
+ */
+static int *distribution_cmp(distribution_t *d1, distribution_t *d2) {
+  int *comparison = calloc(2, sizeof(*comparison));
+  if (comparison == NULL) {
+    return NULL;
+  }
   if (d1 == NULL || d2 == NULL) {
-    return EINVAL;
+    comparison[0] = EINVAL;
+    return comparison;
   }
-
   if (d1->num_buckets != d2->num_buckets) {
-    return EINVAL;
+    comparison[0] = EINVAL;
+    return comparison;
+  }
+  for (size_t i = 0; i < tree_size(d1->num_buckets); i++) {
+    if (d1->tree[i].maximum !=
+        d2->tree[i].maximum) { // will it work with doubles?
```

Comment:
That's a good question. After some googling I found out that a so called epsilon can be used to compare doubles:

"A better way to compare would be fabs(a-b) < epsilon where is epsilon is a small enough value, like 0.000001." (Taken from Reddit)

But I don't know for sure what the best approach is.

Reply:
I agree with octo. I'll leave it like that and add a comment